app/controllers/projects_controller.rb

```diff
@@ -129,6 +129,10 @@ def show_tree_content
 
   def show
     @images = []
+    if @project.blank?
+    redirect_to dashboard_path
```

Comment:
Code indentation isn't proper, and you haven't closed the `if` condition with an `end`. Moreover, this isn't the right way to approach the problem. If the project doesn't exist, we should throw the error right in `return_context` and render a 404.

You've also checked in the `hot_deploy` marker file.